### PR TITLE
Zendesk : name require, identify

### PR DESCRIPTION
**Pull request by sauravlal-rudderstack**
## Description of the change

> For Identify call, a key name is required.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: __tests__/data/zendesk_output.json [hunk 1]
```diff
@@ -1,36 +1,13 @@
...
-        },
-        "XML": {},
-        "JSON_ARRAY": {},
-        "FORM": {}
-      },
-      "files": {},
-      "userId": "297b0750-934b-4411-b66c-9b418cdbc0c9"
+  {
+    "statusCode": 400,
+    "error": "Missing required value from \"name\"",
```
Comment (saikumarrs) [inline __tests__/data/zendesk_output.json:4]: The error message is inappropriate.

It should've been something like this,
```
"Required field "traits.name" is missing"
```
CC @utsabc

Comment (saikumarrs) [reply __tests__/data/zendesk_output.json:4]: This is assuming `name` is being sourced from `traits`. Otherwise, update it accordingly.

Comment (utsabc) [reply __tests__/data/zendesk_output.json:4]:
```suggestion
    "error": "[Zendesk] Missing required value from [traits.name]",
```

Comment (sauravlal-rudderstack) [reply __tests__/data/zendesk_output.json]: @saikumarrs  and @utsabc, Yeah, updated the error message.

File: v0/destinations/zendesk/transform.js [hunk 2]
```diff
@@ -344,6 +344,12 @@ async function processIdentify(message, destinationConfig, headers) {
     destinationConfig,
     "identify"
   );
+  if (!payload.user.name) {
+    throw new CustomError(
+      "[Zendesk] Missing required value from [traits.name]",
+      400
+    );
+  }
```
Comment (saikumarrs) [inline v0/destinations/zendesk/transform.js:352]: We shouldn't be fixing it here. Seems like it needs to be fixed in the common utils.

**Review by saikumarrs — APPROVED** [review]

**Comment by utsabc** [comment]
This code base needs to be revamped

**Review by utsabc — APPROVED** [review]

**Pull request merged**